web3/_utils/proof.py

```diff
@@ -0,0 +1,133 @@
+from eth_utils import (
+    keccak,
+)
+import rlp
+from rlp.sedes import (
+    Binary,
+    big_endian_int,
+)
+
+from web3._utils.encoding import (
+    pad_bytes,
+    pad_hex,
+)
+
+
+def storage_position(map_key_hex, position_hex):
+    '''
+    https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getstorageat
+    Returns the patricia trie key (position) of a mapping's key.
+    This helper function is used for formating arguments to eth_getStorageAt
+    and eth_getProof.
+    @param map_key_hex is the key stored in a map
+    @param position_hex is the position where the mapping variable is defined
+        in the contract
+    '''
+    key = pad_hex(map_key_hex, 256)[2:] + pad_hex(position_hex, 256)[2:]
+    return keccak(bytes.fromhex(key)).hex()
```

Comment:
If this method is to be kept, I generally expect internal APIs to be manipulating bytes instead of hex-encoded strings. The manipulation just looks so much cleaner like:

```py
    key = map_key.rjust(32, b'\0') + position.rjust(32, b'\0')
    return keccak(key)
```

And then we can convert in and out of hex when we need to for user interaction.